### PR TITLE
Fix ruby segment fault

### PR DESCRIPTION
**Pull request by TeBoring**
1) rb_ary_new cannot be called during allocate function. During allocate
fucntion, the containing object hasn't been marked and rb_ary_new may
invoke gc to collect containing object.
2) The global map should be marked before allocating it. Otherwise it
may be garbage collected.

### CODE REVIEW

**Comment by bazel-io** [comment]
Can one of the admins verify this patch?

**Comment by TeBoring** [comment]
Fix #3649

File: ruby/Rakefile [hunk 1]
```diff
@@ -99,7 +99,13 @@ Gem::PackageTask.new(spec) do |pkg|
 end
 
 Rake::TestTask.new(:test => :build) do |t|
-  t.test_files = FileList["tests/*.rb"]
+  t.test_files = FileList["tests/*.rb"].exclude("tests/gc_test.rb")
+end
+
+# gc_test needs to be splited out to ensure the generated file hasn't been
```
Comment (haberman) [inline ruby/Rakefile]: "splited" -> "split"

Comment (TeBoring) [reply ruby/Rakefile]: done

**Comment by joshrose** [comment]
Thanks for this @TeBoring . Any idea when this will be published as part of a new release? Is there somewhere we can look to track what's happening with assembling future releases? Thanks.

**Comment by TeBoring** [comment]
We are working on 3.5.0 release. https://github.com/google/protobuf/milestone/21

**Pull request merged**